BackendWebApp/src/main/java/uk/ac/ebi/biostd/webapp/application/rest/mappers/FileMapper.java

```diff
@@ -24,11 +31,42 @@ public FileDto map(File file, String path) {
     }
 
     public FileDto map(File file, String basePath, String requestPath) {
-        return map(file, getFolderPath(basePath, requestPath) + PATH_SEPARATOR +  file.getName());
+        return map(file, basePath, requestPath, false, "");
+    }
+
+    public FileDto map(File file, String basePath, String requestPath, boolean showArchive, String zipPath) {
+        String path = getFolderPath(basePath, requestPath);
+        path = path.contains(file.getName()) ? path : path + PATH_SEPARATOR + file.getName();
+
+        FileDto mappedFile = map(file, path);
+        if (showArchive && mappedFile.getType() == FileType.ARCHIVE) {
+            mappedFile.setFiles(mapZipInnerFiles(file, basePath, requestPath, zipPath));
+        }
+
+        return mappedFile;
     }
 
     public List<FileDto> map(List<File> files, String basePath, String requestPath) {
-        return files.stream().map(file -> map(file, basePath, requestPath)).collect(Collectors.toList());
+        return map(files, basePath, requestPath, false, "");
+    }
+
+    public List<FileDto> map(List<File> files, String basePath, String requestPath, boolean showArchive, String zipPath) {
+        return files.stream().map(
+                file -> map(file, basePath, requestPath, showArchive, zipPath)).collect(Collectors.toList());
+    }
+
+    @SneakyThrows
+    private List<FileDto> mapZipInnerFiles(File zipFile, String basePath, String requestPath, String zipPath) {
+        File tmpDir = new File(FileUtils.getTempDirectory().getAbsolutePath() + PATH_SEPARATOR + TMP_DIR);
```

Comment:
All this zip file handling looks like a totally indendent concers from mapping beetween objects and dtos, I think we want to create a utility class that do that for us and keep the mapper code simpler.

Reply:
I couldn't agree more, I just wasn't sure about the "Util" classes policy for this project. You know, some people like it, some other don't :man_shrugging: 

I'll do these changes.